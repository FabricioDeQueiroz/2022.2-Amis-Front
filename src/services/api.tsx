/* eslint-disable @typescript-eslint/restrict-template-expressions */
/* eslint-disable no-template-curly-in-string */
import axios from "axios";
import { getUserLocalStorage } from "./auth";

const userLocalStorage = getUserLocalStorage();

const api = axios.create({
  baseURL: "https://service-amis.azurewebsites.net/",
  headers: {
    Authorization: `Bearer ${userLocalStorage?.token}`,
  },
});

export default api;
