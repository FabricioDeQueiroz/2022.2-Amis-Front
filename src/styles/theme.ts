const theme = {
  colors: {
    primary: "#da4d3d",
    white: "#FFF",
    black: "#000",
  },
};

export default theme;
