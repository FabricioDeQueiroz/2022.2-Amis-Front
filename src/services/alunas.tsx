import { AlunasCadastrarDTO } from "../pages/alunas/dtos/AlunasCadastrarDTO";
import api from "./api";

export const cadastraAluna = async (payload: AlunasCadastrarDTO) => {
  return await api.post("/alunas/", payload).then((response: any) => response);
};

export const listaAlunas = async () => {
  return await api.get("/alunas/").then((response: any) => response);
};

export const apagaAluna = async (alunaId: string) => {
  return await api
    .delete("/alunas/" + alunaId)
    .then((response: any) => response);
};

export const editaAluna = async (
  alunaId: string,
  alunaEdit: AlunasCadastrarDTO
) => {
  return await api
    .put("/alunas/" + alunaId, alunaEdit)
    .then((response: any) => response);
};