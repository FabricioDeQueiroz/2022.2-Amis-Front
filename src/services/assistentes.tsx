import { AssistentesCadastrarDTO } from "../pages/assistentes/dtos/AssistentesCadastrarDTO";
import api from "./api";

export const cadastrarAssistente = async (payload: AssistentesCadastrarDTO) => {
  return await api.post("/assistentes/", payload).then((response) => response);
};

export const listarAssistentes = async () => {
  return await api.get("/assistentes/").then((response) => response);
};

export const editarAssistente = async (
  assistenteId: string,
  assistente: Object
) => {
  return await api
    .put("/alunas/" + assistenteId, { assistente })
    .then((response) => response);
};

export const excluirAssistente = async (assistenteId: string) => {
  return await api
    .delete("/alunas/" + assistenteId)
    .then((response) => response);
};
