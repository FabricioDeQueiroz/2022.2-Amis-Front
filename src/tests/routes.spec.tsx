import { QueryClientProvider } from "react-query";
import { BrowserRouter } from "react-router-dom";
import renderer from "react-test-renderer";
import { ThemeProvider } from "styled-components";
import AppRoutes from "../routes";
import { queryClient } from "../services/queryClient";
import theme from "../styles/theme";

describe("Snapshot", () => {
  it("Deve corresponder ao Snapshot", () => {
    const tree = renderer
      .create(
        <QueryClientProvider client={queryClient}>
          <ThemeProvider theme={theme}>
            <AppRoutes />
          </ThemeProvider>
        </QueryClientProvider>
      )
      .toJSON();
    expect(tree).toMatchSnapshot();
  });
});
