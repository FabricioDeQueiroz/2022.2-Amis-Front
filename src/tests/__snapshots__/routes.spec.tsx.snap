// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Snapshot Deve corresponder ao Snapshot 1`] = `
<div>
  <div
    className="sc-hBxehG ibKRjt"
  >
    <div
      className="sc-fnGiBr vpwKY"
    >
      <div
        className="sc-fEXmlR boZxAa"
      >
        <h1
          className="sc-bjfHbI iJdsUp"
        >
          AMIS
        </h1>
        <a
          href="/"
          onClick={[Function]}
        >
          <button
            className="sc-idXgbr jmnPjA"
            onClick={[Function]}
          >
            Home
          </button>
        </a>
        <a
          href="/receitas"
          onClick={[Function]}
        >
          <button
            className="sc-idXgbr dPAtZG"
            onClick={[Function]}
          >
            Receitas
          </button>
        </a>
      </div>
      <a
        href="/login"
        onClick={[Function]}
      >
        <button
          className="sc-kDvujY FpMuN"
          type="submit"
        >
          Área Logada
        </button>
      </a>
    </div>
  </div>
  <div
    className="sc-kgTSHT vNa-dT"
  >
    <div
      className="sc-fLcnxK iuHPaY"
    >
      <h1
        className="sc-iveFHk bNZPJP"
      >
        AMIS
      </h1>
      <h1
        className="sc-iveFHk bNZPJP"
      >
        Mulheres Criativas
      </h1>
      <span
        style={
          {
            "paddingTop": "30px",
          }
        }
      >
        Lorem ipsum dolor sit amet, consectetur adipiscing elit.
      </span>
      <span
        style={
          {
            "paddingBottom": "30px",
          }
        }
      >
        Varius sed pharetra dictum neque massa congue
      </span>
      <div
        style={
          {
            "display": "flex",
            "gap": "15px",
          }
        }
      >
        <button
          className="sc-kDvujY FpMuN"
          type="submit"
        >
          Saiba Mais
        </button>
        <button
          className="sc-kDvujY FpMuN"
          type="submit"
        >
          Faça um pedido
        </button>
      </div>
    </div>
    <div
      style={
        {
          "position": "relative",
        }
      }
    >
      <img
        className="sc-bBABsx jczzxV"
        src="test-file-stub"
        style={
          {
            "bottom": -80,
            "left": -200,
            "position": "absolute",
            "width": "300px",
            "zIndex": 5,
          }
        }
      />
      <img
        className="sc-bBABsx jczzxV"
        src="test-file-stub"
        style={
          {
            "width": "400px",
            "zIndex": 0,
          }
        }
      />
    </div>
  </div>
  <div
    className="sc-cwSeag ePDlRK"
  >
    <h1
      className="sc-lllmON kNZGHC"
    >
      Quem Somos
    </h1>
    <span
      className="sc-iJnaPW fEfPCp"
    >
      Lorem ipsum dolor sit amet, consectetur adipiscing elit. Nullam aliquam egestas ultricies. Morbi venenatis, odio a laoreet vulputate, sapien tellus ornare orci, sit amet feugiat arcu dui sed lorem. Maecenas et lacus vel lacus lacinia porttitor. Nulla tempus dictum efficitur. Suspendisse potenti. Sed ac nulla eget nunc tempor faucibus sed et mauris. Lorem ipsum dolor sit amet, consectetur adipiscing elit. Suspendisse potenti. Quisque tortor ipsum, tincidunt eget ligula at, aliquet commodo purus. Vestibulum ante ipsum primis in faucibus orci luctus et ultrices posuere cubilia curae; In pretium leo ac dolor gravida pretium. Sed eu venenatis lacus, efficitur pulvinar massa. Morbi aliquet sapien vitae purus tincidunt, eget consequat erat iaculis. Donec ut sem ac sapien laoreet congue id et neque. Ut fermentum non libero ut consequat. Quisque elementum velit eget ipsum suscipit rutrum.
    </span>
    <h1
      className="sc-lllmON kNZGHC"
    >
       Nossa missão
    </h1>
    <span
      className="sc-iJnaPW fEfPCp"
    >
      Lorem ipsum dolor sit amet, consectetur adipiscing elit. Nullam aliquam egestas ultricies. Morbi venenatis, odio a laoreet vulputate, sapien tellus ornare orci, sit amet feugiat arcu dui sed lorem. Maecenas et lacus vel lacus lacinia porttitor. Nulla tempus dictum efficitur. Suspendisse potenti. Sed ac nulla eget nunc tempor faucibus sed et mauris. Lorem ipsum dolor sit amet, consectetur adipiscing elit. Suspendisse potenti. Quisque tortor ipsum, tincidunt eget ligula at, aliquet commodo purus. Vestibulum ante ipsum primis in faucibus orci luctus et ultrices posuere cubilia curae; In pretium leo ac dolor gravida pretium. Sed eu venenatis lacus, efficitur pulvinar massa. Morbi aliquet sapien vitae purus tincidunt, eget consequat erat iaculis. Donec ut sem ac sapien laoreet congue id et neque. Ut fermentum non libero ut consequat. Quisque elementum velit eget ipsum suscipit rutrum.
    </span>
  </div>
  <div
    className="sc-gikAfH jwNVoc"
  >
    <div
      className="sc-ezOQGI fPIdKw"
    >
      <h1
        className="sc-bYMpWt bpkZWv"
      />
      <span
        className="sc-kMjNwy hrpIXe"
      >
        Mulheres atendidas
      </span>
    </div>
    <div
      className="sc-ezOQGI fPIdKw"
    >
      <h1
        className="sc-bYMpWt bpkZWv"
      />
      <span
        className="sc-kMjNwy hrpIXe"
      >
        Formações profissionais
      </span>
    </div>
  </div>
  <div
    className="sc-ilhmMj jaunbF"
  >
    <div>
      <h1
        className="sc-jIRcFI eoUPgx"
      >
        Nossas receitas
      </h1>
    </div>
  </div>
  <div
    className="sc-dIfARi hIOfut"
  >
    <a
      href="https://www.instagram.com/amismulherescriativas/"
    >
      <img
        className="sc-hHTYSt ibSsFW"
        src="test-file-stub"
        style={
          {
            "marginBottom": "20px",
            "marginRight": "20x",
            "width": "40px",
          }
        }
      />
    </a>
    <a
      href="https://pt-br.facebook.com/InstitutoMulheresCriativas/"
    >
      <img
        className="sc-hHTYSt ibSsFW"
        src="test-file-stub"
        style={
          {
            "marginBottom": "20px",
            "marginRight": "20px",
            "width": "20px",
          }
        }
      />
    </a>
    <div>
      <span
        className="sc-dmctIk kzFREf"
      >
        © 2022. All rights reserved.
      </span>
    </div>
  </div>
</div>
`;
